### ISSUE TITLE
Balance: Check your EBT balance via SMS

### ISSUE THREAD

**Issue by lippytak** [codeforamerica/project-ideas#34]
Electronic Benefit Transfer (EBT) cards are used to distribute many cash benefits in CA, including CalFresh and local welfare programs. You can check the balance online by going here and creating an account: https://www.ebt.ca.gov/caebtclient/reciplogin_client.jsp. But that's lame. You can also call the # and enter your 16 digit account number: 1-877-328-9677. But that's lame too.

But what if we could use Twilio to call 1-877-328-9677 on your behalf, transcribe the message, and text back the amount? Here's the basic flow:
1. Client texts us account # (just the first time - maybe 4 digit confirmation on subsequent uses?)
2. We initiate call to 1-877-328-9677 and follow the first two prompts (language + case #) automatically
3. We transcribe audio recording and parse the transcription to get the current balance
4. We respond to client with balance

@daguar's hacking on this as we speak to determine feasibility before we figure out potential value and optimal UX.

**Comment by daguar** [comment]
w00t I think Twilio can do the transcription.

I've tested it and — at least based on a sample size of a one account balance ($141.74) — the Twilio robots did their job.

It's worth noting that every transcription is 5 cents, so if we can do it ourselves we can cut down on costs.

**Comment by lippytak** [comment]
It's alive!! https://github.com/daguar/balance

**Comment by fureigh** [comment]
Rapid rapid rapid!

**Comment by lippytak** [comment]
Aaaaaaaaand it works!
![blur](https://cloud.githubusercontent.com/assets/2533112/3823679/a13fd9b0-1d3c-11e4-99a5-dca676c6cbb9.png)

"Production" phone number = **(415) 798-2736** (Edit: You can test it out using this EBT number: **5077190197646090**).

For now PM me, @daguar, or @alanjosephwilliams to get an EBT number to play with. @alanjosephwilliams what are your thoughts on sharing your EBT card # on GH? Risks/benefits/your immediate emotional reaction to the idea?

**Comment by alanjosephwilliams** [comment]
I don't think it can be used unless the card is run—I haven't yet seen a place to enter the number online (next project??). And as you know I'm not particularly risk averse.

So yes: 5077190197646090

Let me know if the second number (which I believe is meant to represent me, rather than my account) is required.

**Comment by ajb** [comment]
y'all are awesome. :star2:

**Comment by daguar** [comment]
First user in the wild (via signs in West SOMA): 3 days, 23 hours after initial idea posted.

**Comment by billypat** [comment]
Hey guys, Our local food Startup wants to accept EBT/Snap, are you guys working on anything like this or know someone that is? 

Cheers,

Will
https://github.com/producerun/
will@producerun.com

**Comment by lippytak** [comment]
hey @hackandgrow! So I hate to do this but I have to refer you to the federal government (bleh): http://www.fns.usda.gov/snap/retailers-0. In order to accept SNAP, FNS needs to authorize you as a retailer. There are some inventory and POS requirements (no online sales). If you choose to go through this process I'd be really interested to hear how it goes. Good luck!

**Comment by billypat** [comment]
Thanks Lippytak, We are going through the process now.
I learnt that EBT is encrypted differently to the credit and debit cards and restrictions on some products

We are going to become a snap retailer but we are looking for solutions to make this a win win for our farmers and the people that we currently have to say no too(we hate saying no).

Any suggestions on how we could get started with a solutions, perhaps we create an open source EBT payment gateway to show government?

**Comment by lippytak** [comment]
You're over my head, Will. Sorry! Hopefully someone else can weigh in...

**Comment by Snapbenefits** [comment]
You can easily check it online via [LINK REDACTED]

Also see gas stations where you can use your EBT card [LINK REDACTED]

**Comment by daguar** [comment]
Well hi there @Snapbenefits! Looks like you're making an SEO play on SNAP info, eh? Care to share any details of what you're up to?

**Comment by daguar** [comment]
Sorry @Snapbenefits but since you haven't replied I've had to remove your links -- can't do link building on this train!

**Comment by Snapbenefits** [comment]
Didn't get your point

**Comment by bmwwoo** [comment]
This looks like it was a pretty creative solution for acquiring the EBT balance! Do you know of any way to get the transaction history or if there's any APIs for this now?

**Comment by Snapbenefits** [comment]
Not sure about that.
